src/pages/interviews.tsx

```diff
@@ -118,6 +118,23 @@ function Applicants({ type, applicants, interviews, refetchInterviews,
   refetchInterviews: () => void,
   refetchApplicants: () => void,
 }) {
+  // A map from applicants' user ids to sources.
+  const [sources, setSources] = useState<{ [id: string]: string }>({});
+  const updateSource = (id: string) => (source: string) => {
+    setSources(prev => {
+      if (prev[id] === source) return prev; 
+      return { ...prev, [id]: source };
+    });
+  };
+  const sortedApplicants = useMemo(() => {
+    return applicants.sort((a1, a2) => {
+      const comp = compareChinese(sources[a1.id], sources[a2.id]);
+      return comp !== 0 ? comp :
+        compareChinese(formatUserName(a1.name, 'formal'), 
+          toPinyin(formatUserName(a2.name, 'formal')));
```

Comment:
this is not correct

Reply:
also, do you really need formatUserName? can you simply compareChinese(a1.name, a2.name)? compareChinese may allow null parameters.